examples/Schnorr.purs

```diff
@@ -0,0 +1,133 @@
+module Ctl.Examples.Schnorr (contract) where
+
+import Contract.Prelude
+
+import Contract.Address (getNetworkId, validatorHashEnterpriseAddress)
+import Contract.Crypto.Schnorr
+  ( SchnorrPublicKey
+  , SchnorrSignature
+  , deriveSchnorrSecp256k1PublicKey
+  , signSchnorrSecp256k1
+  )
+import Contract.Crypto.Utils (randomPrivateKey)
+import Contract.Log (logInfo')
+import Contract.Monad (Contract, liftContractM, liftedE)
+import Contract.PlutusData
+  ( class ToData
+  , PlutusData(Constr)
+  , Redeemer(Redeemer)
+  , toData
+  , unitDatum
+  )
+import Contract.Prim.ByteArray (ByteArray, byteArrayFromIntArrayUnsafe)
+import Contract.ScriptLookups as Lookups
+import Contract.Scripts (Validator, validatorHash)
+import Contract.TextEnvelope (decodeTextEnvelope, plutusScriptV2FromEnvelope)
+import Contract.Transaction
+  ( TransactionHash
+  , awaitTxConfirmed
+  , balanceTx
+  , signTransaction
+  , submit
+  )
+import Contract.TxConstraints as Constraints
+import Contract.Utxos (utxosAt)
+import Contract.Value as Value
+import Data.Map as Map
+import Data.Set as Set
+
+newtype SchnorrRedeemer = SchnorrRedeemer
+  { msg :: ByteArray
+  , sig :: SchnorrSignature
+  , pk :: SchnorrPublicKey
+  }
+
+derive instance Generic SchnorrRedeemer _
+derive instance Newtype SchnorrRedeemer _
+
+instance ToData SchnorrRedeemer where
```

Comment:
Also, maybe we will store Plutus sources along with compiled version for reference, if there are such sources? Cuz last time I debugged Plutus script error it was not very convenient.

Reply:
We're not going to debug these, once done they will never be recompiled